### ISSUE TITLE
[path_provider] since 2.0.12 (using new path_provider_foundation) getApplicationSupportDirectory path changed

### ISSUE THREAD

**Issue by fix92** [flutter/flutter#119133]
## Steps to Reproduce
on iOS, compare the output of getApplicationSupportDirectory for path_provider 2.0.11 vs 2.0.12

```dart
    final dir = await getApplicationSupportDirectory();
    print(dir.path);
```

Probably relevant to this change https://github.com/flutter/plugins/pull/6989

**Expected results:**
Same result as the app relies on the same path when updating this.

**Actual results:**
Example on an iOS simulator:
2.0.11
`/Users/user/Library/Developer/CoreSimulator/Devices/<samedeviceid>/data/Containers/Data/Application/<applicationid1>/Library/Application Support`

2.0.12
`/Users/user/Library/Developer/CoreSimulator/Devices/<samedeviceid>/data/Containers/Data/Application/<applicationid2>/Library/Application Support/com.example.app`

https://github.com/flutter/plugins/blob/main/packages/path_provider/path_provider/CHANGELOG.md#2012

<details>

<summary>Code sample</summary>

```dart
    final dir = await getApplicationSupportDirectory();
    print(dir.path);
```

</details>

<details>
  <summary>Logs</summary>
```
 flutter analyze
Analyzing example-app...                                               
No issues found! (ran in 2.1s)
```

```
[✓] Flutter (Channel stable, 3.7.0, on macOS 12.6 21G115 darwin-arm64, locale en-DE)
    • Flutter version 3.7.0 on channel stable at /Users/user/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (2 days ago), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: /Users/user/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13F100
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/user/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Android Studio (version 2021.2)
    • Android Studio at /Users/user/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8609683/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.2.3)
    • IntelliJ at /Users/user/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/222.4345.14/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Community Edition (version 2022.1.3)
    • IntelliJ at /Users/user/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5921.22/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • iPhone SE (3rd generation) (mobile) • FAF92996-F1BE-405B-A382-2017897DA696 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-5 (simulator)
    • macOS (desktop)                     • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm64
    • Chrome (web)                        • chrome                               • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by stuartmorgan** [comment]
Thanks for reporting this; it was definitely not intentional.

**Comment by stuartmorgan** [comment]
This is fixed in `path_provider_foundation` 2.1.1, and 2.1.0 has been retracted. To get the fix, `flutter pub upgrade`.
